app/models/consultation.rb

```diff
@@ -49,6 +49,11 @@ class AddNeighbourAddressesError < StandardError; end
     has_many :neighbour_responses
   end
 
+  has_many :reviews, as: :owner, dependent: :destroy
+  has_many :neighbour_reviews, -> { where(specific_attributes: {consultation_type: "neighbour"}) }, as: :owner, class_name: "Review", dependent: :destroy
```

Comment:
I'd move that `where` to a scope on review and then you can just call that in the proc here, e.g.

``` ruby
with_options as: :owner do
  has_many :reviews, dependent: :destroy
  has_many :neighbour_reviews, -> { neighbour_reviews }, class_name: "Review"
end
```

Note when you're associating to the same table multiple times that you should only have one association with a `dependent: :destroy` option otherwise Active Record will try to load and destroy the the same objects multiple times.

Depending on the expected number of reviews you might be better off using association extension methods, e.g. 

``` ruby
has_many :reviews, as: :owner, dependent: :destroy do
  def neighbours
    select(&:neighbour?)
  end

  def latest_neighbour
    neighbours.sort_ by(&:created_at).last
  end
end
```

That way there's a single set of review objects in memory and you don't run the risk of having multiple instances of a review where one is loaded via `reviews` and another loaded by `neighbour_reviews`.